guix/cando.scm

```diff
@@ -0,0 +1,101 @@
+                (string-append source-dir "/dependencies")
+                (string-append source-dir "/extensions")))
+         (git-dirs
+          (apply append
+                 (list (string-append source-dir "/src/bdwgc")
+                       (string-append source-dir "/src/libatomic_ops")
+                       (string-append source-dir "/src/lisp/modules/asdf")
+                       ;; This is a special case: the parent is omitted
+                       ;; from external-dirs, but added to descend-dirs.
+                       (string-append source-dir "/extensions/seqan-clasp/seqan"))
+                 (map-in-order scandir-absolute external-dirs)))
+         (descend-dirs `(,@git-dirs
+                         ,(string-append source-dir "/extensions/seqan-clasp")
+                         ,@external-dirs))
+         (predicates (map-in-order git-predicate (cons source-dir git-dirs))))
+    (package
+      (inherit clasp-cl)
+      (name "cando")
+      (version version)
+      (source
+       (local-file source-dir (git-file-name "cando" version)
+                   #:recursive? #t
+                   #:select?
+                   (lambda (file stat)
+                     (or (any (lambda (dir) (string=? dir file)) descend-dirs)
+                         (any (lambda (pred) (pred file stat)) predicates)))))
+      (arguments
+       (substitute-keyword-arguments (package-arguments clasp-cl)
+         ((#:configure-flags flags '())
+          #~(cons*
+             "--extensions=cando,seqan-clasp"
+             ;; Had trouble simplifying this using find.
+             ;; This is prepended because Koga keeps the first instance
+             ;; of repeated options.
+             (string-append
+              "--ldflags="
+              "-Wl,-rpath," #$(this-package-input "clang-toolchain") "/lib"
+              " -Wl,-rpath," (ungexp (this-package-input "gcc") "lib") "/lib"
+              " -Wl,-rpath," #$(this-package-input "fmt") "/lib"
+              " -Wl,-rpath," #$(this-package-input "gmp") "/lib"
+              " -Wl,-rpath," #$(this-package-input "libelf") "/lib"
+              " -Wl,-rpath," #$(this-package-input "expat") "/lib"
+              " -Wl,-rpath," #$(this-package-input "netcdf") "/lib")
+             #$flags))
+         ((#:phases phases #~%standard-phases)
+          #~(modify-phases #$phases
+              ;; Fixed in
+              ;; <https://github.com/cando-developers/cando/commit/85ffab474faa342c38ced5b84553db035d23687b>.
+              (add-after 'unpack 'fix-build
+                (lambda _
+                  (substitute*
+                      '("extensions/cando/src/geom/matrix.cc"
+                        "extensions/cando/src/geom/ovector3.cc"
+                        "extensions/cando/src/geom/vector3.cc"
+                        "extensions/cando/src/kinematics/bondedJoint.cc")
+                    (("\\(isnan\\(") "(std::isnan(")
+                    (("\\|isnan\\(") "|std::isnan("))))))))
```

Comment:
Do we still need these substitutions?

Reply:
You're right, we don't. I hadn't realized that repos.sexp followed Cando's main branch.

Reply:
Yes, When we do a release the repos.sexp in the source archive will have a specific commit for all dependencies.